src/icrar/leap-accelerate/ms/MeasurementSet.cc

```diff
@@ -32,30 +34,102 @@ namespace icrar
+        
+        if(antennas.size() != GetNumStations())
+        {
+            LOG(error) << "unique antennas does not match number of stations";
+            LOG(error) << "unique antennas: " << antennas.size();
+            LOG(error) << "stations: " << GetNumStations();
+            //throw exception("number of stations incorrect", __FILE__, __LINE__);
+        }
 
-    // MeasurementSet::MeasurementSet(std::istream& stream, boost::optional<int> overrideNStations)
-    // {
-    //     // don't skip the whitespace while reading
-    //     std::cin >> std::noskipws;
+        //Baselines
+        //Validate number of baselines in first epoch
+        casacore::Vector<double> time = m_msmc->time().getColumn();
+        auto epoch = time[0];
+        uint32_t epochRows = 0;
+        for(size_t i = 0; i < time.size(); i++)
+        {
+            if(time[i] == epoch) epochRows++;
+        }
 
-    //     // use stream iterators to copy the stream to a string
-    //     std::istream_iterator<char> it(std::cin);
-    //     std::istream_iterator<char> end;
-    //     std::string results = std::string(it, end);
-        
-    //     m_stations = overrideNStations.is_initialized() ? overrideNStations.get() : m_measurementSet->antenna().nrow();
-    // }
+        if(epochRows != GetNumBaselines())
+        {
+            LOG(warning) << "epoch rows does not match baselines";
+            LOG(warning) << "epoch rows: " << epochRows;
+            LOG(warning) << "baselines: " << GetNumBaselines();
+            throw exception("epoch size doesnt match number of baselines", __FILE__, __LINE__);
+        }
+
+        if(GetNumRows() < GetNumBaselines())
+        {
+            std::stringstream ss;
+            ss << "invalid number of rows, expected >=" << GetNumBaselines() << ", got " << GetNumRows();
+            throw icrar::file_exception(GetFilepath().get_value_or("unknown"), ss.str(), __FILE__, __LINE__);
+        }
+
+        if(GetNumRows() % GetNumBaselines() != 0)
+        {
+            LOG(warning) << "number of rows not an integer multiple of number of baselines";
+            LOG(warning) << "baselines: " << GetNumBaselines()
+                         << " rows: " << GetNumRows()
+                         << "total epochs ~= " << (double)GetNumRows() / GetNumBaselines();
+            throw exception("number of rows not an integer multiple of baselines", __FILE__, __LINE__);
+        }
+
+        //Flags
+        auto epochIndices = casacore::Slice(0, GetNumBaselines(), 1);
+        auto flagSlice = casacore::Slicer(
+            casacore::IPosition(3,0,0,0),
+            casacore::IPosition(3,1,1,GetNumBaselines()),
+            casacore::IPosition(3,1,1,1));
+
+        casacore::Vector<bool> fg = m_msmc->flag().getColumn()
+        (flagSlice).reform(casacore::IPosition(1, GetNumBaselines()))
+        (epochIndices);
+
+        if(fg.size() != GetNumBaselines())
```

Comment:
This will always be `true` by construction (you are reading those many flags with the given `flagSlice`) so it doesn't make sense to check (maybe assert if you want?). If there's a problem reading those many flags the casacore will already throw an exception.

Also, `fg` doesn't seem to be otherwise used, so the whole read operation could be avoided.

Reply:
I'll take this out as I wanted to add flagging in a seperate PR to help revert or debug in https://github.com/ICRAR/leap-accelerate/pull/66